src/app/pages/ConnectorDetailsPage/ConfigurationTab/CommonStep.tsx

```diff
@@ -37,44 +37,59 @@ export const CommonStep: FC<CommonStepProp> = ({
 
   const [isSAUpdate, setIsSAUpdate] = React.useState(false);
   const [passwordHidden, setPasswordHidden] = React.useState<boolean>(true);
-
+  const [showEyeIcon, setShowEyeIcon] = React.useState<boolean>(false);
   const onNameChange = (val: any) => {
     onUpdateConfiguration('common', { ...configuration, name: val });
     val === '' ? changeIsValid(false) : changeIsValid(true);
   };
 
-  const onSAChange = (
-    val: string,
-    event: React.FormEvent<HTMLInputElement>
-  ) => {
-    if (event.currentTarget.id === 'connector-sa-id') {
-      onUpdateConfiguration('common', {
-        ...configuration,
-        service_account: {
-          client_id: val,
-          client_secret: configuration.service_account.client_secret,
-        },
+  const onSAChange = (clientId: string) => {
+    onUpdateConfiguration('common', {
+      ...configuration,
+      service_account: {
+        client_id: clientId,
+        client_secret: configuration.service_account.client_secret,
+      },
+    });
+    !isSAUpdate &&
+      setIsSAUpdate((prev) => {
+        if (!prev) {
+          changeIsValid(false);
+        }
+        return true;
       });
-      !isSAUpdate &&
-        setIsSAUpdate((prev) => {
-          if (!prev) {
-            changeIsValid(false);
-          }
-          return true;
-        });
+
+    if (
+      configuration.service_account.client_secret.includes('*') ||
+      clientId === ''
+    ) {
+      setShowEyeIcon(false);
+      changeIsValid(false);
     } else {
-      onUpdateConfiguration('common', {
-        ...configuration,
-        service_account: {
-          client_id: configuration.service_account.client_id,
-          client_secret: val,
-        },
-      });
+      setShowEyeIcon(true);
+      changeIsValid(true);
     }
-
-    val === '' ? changeIsValid(false) : changeIsValid(true);
   };
 
+  const onSASecretChange = (secret: string) => {
+    onUpdateConfiguration('common', {
+      ...configuration,
+      service_account: {
+        client_id: configuration.service_account.client_id,
+        client_secret: secret,
+      },
+    });
+    if (
+      configuration.service_account.client_secret.includes('*') ||
```

Comment:
Maybe (related to the previous question) compare against the placeholder value instead?